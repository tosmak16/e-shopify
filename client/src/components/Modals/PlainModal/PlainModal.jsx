import { useState, useEffect } from 'react';

import styles from './PlainModal.scss';
import { iconClose } from '../../../assets/icons';

const PlainModal = props => {
  const { children, shouldModalBeVisible, exposeIsModalVisible, isCloseIconVisible } = props;
  const [isModalVisible, setIsModalVisible] = useState(shouldModalBeVisible);
  const handleCloseModalViaOverLayClose = ({ stopPropagation, target: { id } }) => {
    if (id !== 'modal') {
      return stopPropagation;
    }
    exposeIsModalVisible(false);
    return setIsModalVisible(false);
  };

  const handleCloseModalViaCloseIcon = () => {
    exposeIsModalVisible(false);
    return setIsModalVisible(false);
  };

  const modalVisibilityClass = isModalVisible ? 'modal--visible' : 'modal--hidden';

  useEffect(() => {
    setIsModalVisible(shouldModalBeVisible);
  }, [shouldModalBeVisible]);

  return (
    <div
      styleName={`modal ${modalVisibilityClass}`}
      id="modal"
      role="presentation"
      onClick={handleCloseModalViaOverLayClose}
    >
      <div styleName="modal__container" id="modal__container">
        {isCloseIconVisible && (
          <div
            styleName="modal__close-icon"
            role="presentation"
            onClick={handleCloseModalViaCloseIcon}
          >
            <img src={iconClose} alt="Close" />
          </div>
        )}
        {children}
      </div>
    </div>
  );
};

PlainModal.propTypes = {
  shouldModalBeVisible: PropTypes.bool,
  children: PropTypes.node,
  exposeIsModalVisible: PropTypes.func,
  isCloseIconVisible: PropTypes.bool
};

PlainModal.defaultProps = {
  shouldModalBeVisible: false,
  children: null,
  exposeIsModalVisible: () => {},
  isCloseIconVisible: false
};

export default CSSModules(PlainModal, styles, { allowMultiple: true });
