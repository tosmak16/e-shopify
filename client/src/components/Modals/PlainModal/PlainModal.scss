@import "~variables";

.modal {
    position: fixed;
    top: 0;
    left: 0;
    background: rgba(196, 196, 196, 0.7);
    height: 100%;
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    z-index: 1000;
    cursor: pointer;
    transition: opacity 0.2s ease;

    &__container {
        background: white;
        color: black;
        display: flex;
        flex-direction: column;
        transition: all 0.2s ease;

        @media (max-width: 1024px) {
            height: 100%;
            width: 100%;
        }

    }

    &__close-icon {
        justify-self: end;
        align-self: flex-end;
        padding: 20px;
        height: 10px;
        cursor: pointer;
    }

    &--visible {
        opacity: 1;
        visibility: visible;

        .modal__container {
            transform: scale(1, 1);
        }
    }

    &--hidden {
        opacity: 0;
        visibility: hidden;

        .modal__container {
            transform: scale(0, 0);
        }
    }
}