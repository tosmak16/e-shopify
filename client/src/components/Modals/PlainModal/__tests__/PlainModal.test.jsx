import PlainModal from '../PlainModal';

describe('PlainModal', () => {
	it('renders without close icon', () => {
		const tree = shallow(
			<PlainModal>hello</PlainModal>);
		expect(tree).toMatchSnapshot();
		expect(tree.find('.modal__close-icon').length).toBe(0);
	});

	it('renders with close icon', () => {
		const tree = shallow(
			<PlainModal isCloseIconVisible />);
		expect(tree).toMatchSnapshot();
		expect(tree.find('.modal__close-icon').length).toBe(1);
	});

	it('renders with modal--visible class when shouldModalBeVisible is true', () => {
		const tree = shallow(
			<PlainModal shouldModalBeVisible />);
		expect(tree).toMatchSnapshot();
		expect(tree.find('.modal--visible').length).toBe(1);
	});

	it('renders without modal--hidden class when shouldModalBeVisible is false', () => {
		const tree = shallow(
			<PlainModal>hello</PlainModal>);
		expect(tree.find('.modal--hidden').length).toBe(1);
	});

	it('should set isModalVisible to false when modal__close-icon is clicked', () => {
		let isModalVisible = true;
		const tree = shallow(
			<PlainModal
				isCloseIconVisible
				shouldModalBeVisible
				exposeIsModalVisible={(status) => { isModalVisible = status; }}
			/>);
		tree.find('.modal__close-icon').simulate('click');
		expect(isModalVisible).toBeFalsy();
	});

	it('should set isModalVisible to false when modal is clicked', () => {
		let isModalVisible = true;
		const tree = shallow(
			<PlainModal
				isCloseIconVisible
				shouldModalBeVisible
				exposeIsModalVisible={(status) => { isModalVisible = status; }}
			/>);
		tree.find('.modal').simulate('click', { stopPropagation: () => undefined, target: { id: '.modal' } });
		tree.find('.modal').simulate('click', { stopPropagation: () => undefined, target: { id: 'modal' } });
		expect(isModalVisible).toBeFalsy();
	});
	it('should maitain isModalVisible to be true when modal__container is clicked', () => {
		let isModalVisible = true;
		const tree = shallow(
			<PlainModal
				isCloseIconVisible
				shouldModalBeVisible
				exposeIsModalVisible={(status) => { isModalVisible = status; }}
			/>);
		tree.find('.modal__container').simulate('click', { target: { id: '.modal__container' } });
		expect(isModalVisible).toBeTruthy();
	});
});
