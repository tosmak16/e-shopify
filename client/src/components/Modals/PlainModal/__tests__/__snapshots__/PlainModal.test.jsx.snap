// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlainModal renders with close icon 1`] = `
<div
  className="modal modal--hidden"
  id="modal"
  onClick={[Function]}
  role="presentation"
>
  <div
    className="modal__container"
    id="modal__container"
  >
    <div
      className="modal__close-icon"
      onClick={[Function]}
      role="presentation"
    >
      <img
        alt="Close"
        src="test-file-stub"
      />
    </div>
  </div>
</div>
`;

exports[`PlainModal renders with modal--visible class when shouldModalBeVisible is true 1`] = `
<div
  className="modal modal--visible"
  id="modal"
  onClick={[Function]}
  role="presentation"
>
  <div
    className="modal__container"
    id="modal__container"
  />
</div>
`;

exports[`PlainModal renders without close icon 1`] = `
<div
  className="modal modal--hidden"
  id="modal"
  onClick={[Function]}
  role="presentation"
>
  <div
    className="modal__container"
    id="modal__container"
  >
    hello
  </div>
</div>
`;
